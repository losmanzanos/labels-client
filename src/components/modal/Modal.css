/* The Modal (background) */
.modal {
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgb(0, 0, 0);
  background-color: rgba(0, 0, 0, 0.4);
}

p {
  color: black;
}

/* Modal Content/Box */
.modal-content {
  background-color: #fefefe;
  margin: 15% auto;
  padding: 20px;
  border: 1px solid #888;
  width: 80%;
}

.modal-image {
  width: 200px;
  height: auto;
  margin: auto;
  text-align: center;
}

/* The Close Button */
.close {
  color: #aaa;
  float: right;
  font-size: 28px;
  font-weight: bold;
}

.close:hover,
.close:focus {
  color: black;
  text-decoration: none;
  cursor: pointer;
}
