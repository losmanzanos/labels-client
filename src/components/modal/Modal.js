import React, { useEffect, useState } from "react";
import "./Modal.css";
import imageAPIService from "../../services/image-api-service";

const Modal = (props) => {
  const [confirm, setConfirm] = useState(false);

  const [features, setFeatures] = useState([]);
  useEffect(() => {
    if (!props.show.id) {
      return;
    }
    imageAPIService
      .getFeatures(props.show.id)
      .then((data) => setFeatures(data));
  }, [props.show.id]);

  if (!props.show) {
    return null;
  }

  const closeModal = () => {
    props.setShow(false);
    setConfirm(false);
  };

  const deleteImage = () => {
    imageAPIService.deleteImage(props.show.id).then(() => {
      props.getData();
      closeModal();
    });

    console.log("Deleted!");
  };
  console.log(features);

  return (
    <div className="modal">
      <div className="modal-content">
        <span className="close" onClick={closeModal}>
          &times;
        </span>
        <img className="modal-image" src={props.show.url} />
        {features.map((feature) => (
          <p key={feature.id}>
            Label:{" "}
            {feature.label.replace(/(?:^|\s)\S/g, function (a) {
              return a.toUpperCase();
            })}
          </p>
        ))}
        <br />
        {confirm ? (
          <>
            <button onClick={deleteImage}>Confirm</button>
            <button onClick={() => setConfirm(false)}>Cancel</button>
          </>
        ) : (
          <button onClick={() => setConfirm(true)}>Delete</button>
        )}
      </div>
    </div>
  );
};

export default Modal;
