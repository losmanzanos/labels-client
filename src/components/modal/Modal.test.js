import renderer from "react-test-renderer";

//Smoke Test

// make React available
import React from "react";

// make the ReactDOM available, necessary for rendering the component
import ReactDOM from "react-dom";

// make the App component available
import Modal from "./Modal";

// this is the test case
it("renders without crashing", () => {
  // first create a DOM element to render the component into
  const div = document.createElement("div");

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<Modal />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

//Snapshot

it("renders the UI as expected", () => {
  const tree = renderer.create(<Modal />).toJSON();
  expect(tree).toMatchSnapshot();
});
