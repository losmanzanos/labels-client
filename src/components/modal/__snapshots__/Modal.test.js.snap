// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="modal"
>
  <div
    className="modal-content"
  >
    <span
      className="close"
      onClick={[Function]}
    >
      Ã—
    </span>
    <img
      className="modal-image"
    />
    <br />
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
